packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1185,7 +1185,10 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
 
       renderObject.debugChildIntegrityEnabled = false; // Moving children will temporary violate the integrity.
       newChildren.keys.forEach(processElement);
-      if (_didUnderflow) {
+      final int newLastIndex = _childElements.lastKey() ?? -1;
+      // We don't worry about underflow if the last child is not longer at the
```

Comment:
Why do we not worry about underflow in this condition?

Reply:
I am not entirely sure about the reason why we have the underflow logic, I can only guess by how it is used.

It seems like render sliver list set it to true when it think it has reached the end of the child list. Thus, if the children are reordered, the previous last child is moved to other place, and we no longer at the end of the child list, we should ignore this request.

Am I missing something?

Reply:
underflow is there to ensure children rebuild will update max scroll offset

for example if you have a list of 10 items, and you scrolled to the end of the list.
If we add an item to the end of the list, the sliver element rebuild will not change its child elements because it will only update its existing children. Since the children does not change during the rebuild, we will skip layout phase. Thus the max scroll offset will not get update, user will not be able to scroll to the 11th child.

That is why we proactively build one more child at the end to ensure the max scroll offset is updated.

This logic is not needed when "any" of the children gets updated, because it will trigger a layout. I will update the logic